tests/test_private_attributes.py

```diff
@@ -160,8 +160,8 @@ def __init__(self):
     m = Model()
     for attr in Model.__slots__:
         assert object.__getattribute__(m, attr) == 'spam'
-        with pytest.raises(ValueError, match=f'"Model" object has no field "{attr}"'):
-            setattr(m, attr, 'not spam')
+        setattr(m, attr, 'not spam')
+        assert object.__getattribute__(m, attr) == 'not spam'
```

Comment:
@MrMrRobat I added both computed fields and raw descriptors support for `__set__`.
IMO it's consistent with a raw class but I'd be happy to have your point of view

Reply:
Thanks for mentioning! At first glance, I don't quite understand why the behaviour changed there. I will take a closer look at this later today or in the next few days.

Reply:
It's due to my last commit 2ff0541

Reply:
Hm, I see now, thanks. Looks like `if name in self.__private_attributes__:` line in `BaseModel.__setattr__` is redundant now. Not sure if it should be deleted though.

Reply:
Yep with the last commit to handle properly raw properties, the special cases for private attributes and computed fields become "almost" redundant. But I feel like we should keep them untouched for debugging and making the whole code more explicit